airbyte-webapp/src/components/connectorBuilder/types.ts

```diff
@@ -607,3 +674,246 @@ export const convertToManifest = (values: BuilderFormValues): ConnectorManifest
     spec,
   };
 };
+
+function manifestPrimaryKeyToBuilder(manifestStream: DeclarativeStream): BuilderStream["primaryKey"] {
+  if (!isEqual(manifestStream.primary_key, manifestStream.retriever.primary_key)) {
+    throw new ManifestCompatibilityError(
+      manifestStream.name,
+      "primary_key is not consistent across stream and retriever levels"
+    );
+  }
+  if (manifestStream.primary_key === undefined) {
+    return [];
+  } else if (Array.isArray(manifestStream.retriever.primary_key)) {
+    if (manifestStream.retriever.primary_key.length > 0 && Array.isArray(manifestStream.retriever.primary_key[0])) {
+      throw new ManifestCompatibilityError(manifestStream.stream.name, "primary_key contains nested arrays");
+    } else {
+      return manifestStream.retriever.primary_key as string[];
+    }
+  } else {
+    return [manifestStream.retriever.primary_key];
+  }
+}
+
+function manifestPaginatorToBuilder(
+  manifestPaginator: SimpleRetrieverPaginator | undefined,
+  streamName: string | undefined,
+  globalUrlBase: string
+): BuilderPaginator | undefined {
+  if (manifestPaginator === undefined || manifestPaginator.type === "NoPagination") {
+    return undefined;
+  }
+
+  if (manifestPaginator.page_token_option === undefined) {
+    throw new ManifestCompatibilityError(streamName, "paginator does not define a page_token_option");
+  }
+
+  if (manifestPaginator.url_base !== globalUrlBase) {
+    throw new ManifestCompatibilityError(streamName, "paginator.url_base does not match the first stream's url_base");
+  }
+
+  return {
+    strategy: manifestPaginator.pagination_strategy,
+    pageTokenOption: manifestPaginator.page_token_option,
+    pageSizeOption: manifestPaginator.page_size_option,
+  };
+}
+
+function manifestSchemaLoaderToBuilderSchema(
+  manifestSchemaLoader: DeclarativeStreamSchemaLoader | undefined,
+  streamName: string | undefined
+): BuilderStream["schema"] {
+  if (manifestSchemaLoader === undefined) {
+    return undefined;
+  }
+
+  if (manifestSchemaLoader.type !== "InlineSchemaLoader") {
+    throw new ManifestCompatibilityError(
+      streamName,
+      "schema_loader type is unsupported. Only InlineSchemaLoader is supported."
+    );
+  }
+
+  const inlineSchemaLoader = manifestSchemaLoader as InlineSchemaLoader;
+
+  if (inlineSchemaLoader.schema) {
+    return JSON.stringify(inlineSchemaLoader.schema);
```

Comment:
should use nicely formatted schema:
<img width="1051" alt="Screenshot 2023-01-11 at 10 58 16" src="https://user-images.githubusercontent.com/1508364/211782801-267419ec-2ddb-4ff1-8b4f-fc306414d0f3.png">